packages/core/src/render3/interfaces/view.ts

```diff
@@ -826,3 +828,190 @@ export type TData =
+   */
+  readonly html: string;
+
+  /**
+   * The host element to which this `LView` is attached.
+   */
+  readonly hostHTML: string|null;
+
+  /**
+   * Child `LView`s
+   */
+  readonly childViews: Array<LViewDebug|LContainerDebug>;
+
+  /**
+   * Sub range of `LView` containing decls (DOM elements).
+   */
+  readonly decls: LViewRange;
+
+  /**
+   * Sub range of `LView` containing vars (bindings).
+   */
+  readonly vars: LViewRange;
+
+  /**
+   * Sub range of `LView` containing i18n (translated DOM elements).
+   */
+  readonly i18n: LViewRange;
+
+  /**
+   * Sub range of `LView` containing expando (used by DI).
+   */
+  readonly expando: LViewRange;
+}
+
+/**
+ * Human readable version of the `LContainer`
+ *
+ * `LContainer` is a data structure used internally to keep track of child views. The `LContainer`
+ * is designed for efficiency and so at times it is difficult to read or write tests which assert on
+ * its values. For this reason when `ngDevMode` is true we patch a `LContainer.debug` property which
+ * points to `LContainerDebug` for easier debugging and test writing. It is the intent of
+ * `LContainerDebug` to be used in tests.
+ */
+export interface LContainerDebug {
+  readonly native: RComment;
+  /**
+   * Child `LView`s.
+   */
+  readonly views: LViewDebug[];
+  readonly parent: LViewDebug|null;
+  readonly movedViews: LView[]|null;
+  readonly host: RElement|RComment|LView;
+  readonly next: LViewDebug|LContainerDebug|null;
+  readonly hasTransplantedViews: boolean;
+}
+
+
+
+/**
+ * `LView` is subdivided to ranges where the actual data is stored. Some of these ranges such as
+ * `decls` and `vars` are known at compile time. Other such as `i18n` and `expando` are runtime only
+ * concepts.
+ */
+export interface LViewRange {
+  /**
+   * The starting index in `LView` where the range begins. (Inclusive)
+   */
+  start: number;
+
+  /**
+   * The ending index in `LView` where the range ends. (Exclusive)
+   */
+  end: number;
+
+  /**
+   * The length of the range
+   */
+  length: number;
+
+  /**
+   * The merged content of the range. `t` contains data from `TView.data` and `l` contains `LView`
+   * data at an index.
+   */
+  content: LViewRangeContent[];
+}
+
+/**
+ * For convenience the static and instance portions of `TView` and `LView` are merged into a single
+ * object in `LViewRange`.
+ */
+export interface LViewRangeContent {
+  /**
+   * Index into original `LView` or `TView.data`.
+   */
+  index: number;
+
+  /**
+   * Value from the `TView.data[index]` location.
+   */
+  t: any;
+
+  /**
+   * Value from the `LView[index]` location.
+   */
+  l: any;
+}
+
+
+/**
+ * A logical node which comprise into `LView`s.
+ *
+ */
+export interface DebugNode {
+  /**
+   * HTML representation of the node.
+   */
+  html: string|null;
+
+  /**
+   * Human readable node type.
+   */
+  type: 'Container'|'Projection'|'View'|'Element'|'ElementContainer'|'IcuContainer';
```

Comment:
I think this should work here:
```suggestion
  type: typeof TNodeTypeAsString[number];
```

Reply:
Ohh, that is cool. Did not know that. Thanks for showing me a new trick.